src/os/alert/alertmanager.js

```diff
@@ -150,28 +157,34 @@ export default class AlertManager extends EventTarget {
     const throttleMap = this.throttleMap_ && this.throttleMap_[alert] ? this.throttleMap_[alert][severity] : undefined;
     const throttleCount = throttleMap ? throttleMap.length : undefined;
 
-    // fire off the alert
-    const alertEvent = new AlertEvent(alert, severity, opt_limit, opt_dismissDispatcher, throttleCount);
-    this.savedAlerts_.add(alertEvent);
-    this.dispatchEvent(alertEvent);
-
-    // write the message to the logger if defined
-    if (opt_logger != null) {
-      if (throttleMap && throttleMap.length > 1) {
-        alert = `${alert} (${throttleMap.length})`;
-      }
-      switch (severity) {
-        case AlertEventSeverity.ERROR:
-          log.error(opt_logger, alert);
-          break;
-        case AlertEventSeverity.WARNING:
-          log.warning(opt_logger, alert);
-          break;
-        case AlertEventSeverity.INFO:
-        case AlertEventSeverity.SUCCESS:
-        default:
-          log.info(opt_logger, alert);
-          break;
+    if (this.mostRecentAlert_ && this.mostRecentAlert_.getMessage() == alert) {
```

Comment:
I don't think the alert manager needs to handle deduplication beyond what it's already doing with the throttle. The throttle consolidates alerts received within a short window (default 500ms) before they're even dispatched, and from there UI's can be responsible for how they handle further duplicates. Once an event is fired, it shouldn't be changed because UI's would have to know how to respond to that change. It's simpler if they only have to respond to new events and can assume every new event is for new messages.